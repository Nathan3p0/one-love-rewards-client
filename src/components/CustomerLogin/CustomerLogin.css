.main__content-user_lookup {
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: #CF5C36;
}

.customerLoginForm {
    background-color: rgb(255, 255, 255);
    padding: 5rem;
    width: 60%;
}

input[type="tel"]{
    width: 100%;
    border: .2rem solid rgba(0, 0, 0, 0.3);
    border-radius: .3rem;
    font-size: 1.2em;
    margin-right: .2rem;
    margin-top: 1rem;
}

input[type="tel"]:hover {
    cursor: pointer;
    border: .2rem solid rgba(0, 162, 255, 0.849);
}

label {
    font-size: 1.5rem;
    font-weight: 500;
    margin: 2rem 0;
}

button {
    position: relative;
    display: inline-block;
    cursor: pointer;
    outline: none;
    border: 0;
    vertical-align: middle;
    text-decoration: none;
    font-size: inherit;
    font-family: inherit;
}

button.cust-btn {
    width: 100%;
    margin-top: 2rem;
    transition: all 0.5s cubic-bezier(0.65, -0.25, 0.25, 1.95);
    font-weight: 900;
    color: #202121;
    padding: 1.25rem 2rem;
    background: #FFE74C;
    text-transform: uppercase;
}

button.cust-btn:hover,
button.cust-btn:focus,
button.cust-btn:active {
    letter-spacing: 0.125rem;
}