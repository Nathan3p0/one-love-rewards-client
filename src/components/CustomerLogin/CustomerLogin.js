import React, { Component } from 'react'
import './CustomerLogin.css'

class CustomerLogin extends Component {
    static defaultProps = {
        onSearchSuccess: () => { }
    }

    state = {
        error: null
    }

    handlePhoneSubmit = (e) => {
        e.preventDefault()
        console.log('I am working')

        this.props.onSearchSuccess()
    }

    render() {
        return (
            <section className="main__content-user_lookup">
                <div>
                    <h2>View What Rewards You've Earned</h2>
                    <form onSubmit={this.handlePhoneSubmit}>
                        <label htmlFor="phone">Enter your phone number:</label>
                        <input type="tel" id="phone" name="phone" pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
                            placeholder="123-456-7890" required />
                        <button type="submit">Submit</button>
                    </form>
                </div>
            </section>
        )
    }
}

export default CustomerLogin