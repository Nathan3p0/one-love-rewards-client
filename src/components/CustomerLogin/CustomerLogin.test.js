import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import CustomerLogin from './CustomerLogin'

describe('Member List Item Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<CustomerLogin />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<CustomerLogin />).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('renders to UI as expected with PROPS', () => {
        const tree = renderer.create(<CustomerLogin
            onSearchSuccess={() => { }}
        />).toJSON();
        expect(tree).toMatchSnapshot()
    })

})