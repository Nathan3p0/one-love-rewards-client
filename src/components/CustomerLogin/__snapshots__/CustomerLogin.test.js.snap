// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer Login Component Rendering Tests renders to UI as expected 1`] = `
<section
  className="main__content-user_lookup"
>
  <div
    className="cover"
  >
    <form
      className="customerLoginForm"
      onSubmit={[Function]}
    >
      <h2>
        View What Rewards You've Earned
      </h2>
      <div
        role="alert"
      />
      <div>
        <label
          htmlFor="phone"
        >
          Enter your phone number:
        </label>
        <input
          id="phone"
          name="phone_number"
          onChange={[Function]}
          pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
          placeholder="734-555-0505"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div>
        <button
          className="cust-btn"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</section>
`;

exports[`Customer Login Component Rendering Tests renders to UI as expected with PROPS 1`] = `
<section
  className="main__content-user_lookup"
>
  <div
    className="cover"
  >
    <form
      className="customerLoginForm"
      onSubmit={[Function]}
    >
      <h2>
        View What Rewards You've Earned
      </h2>
      <div
        role="alert"
      />
      <div>
        <label
          htmlFor="phone"
        >
          Enter your phone number:
        </label>
        <input
          id="phone"
          name="phone_number"
          onChange={[Function]}
          pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
          placeholder="734-555-0505"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div>
        <button
          className="cust-btn"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</section>
`;

exports[`Member List Item Component Rendering Tests renders to UI as expected 1`] = `
<section
  className="main__content-user_lookup"
>
  <div
    className="cover"
  >
    <form
      className="customerLoginForm"
      onSubmit={[Function]}
    >
      <h2>
        View What Rewards You've Earned
      </h2>
      <div
        role="alert"
      />
      <div>
        <label
          htmlFor="phone"
        >
          Enter your phone number:
        </label>
        <input
          id="phone"
          name="phone_number"
          onChange={[Function]}
          pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
          placeholder="734-555-0505"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div>
        <button
          className="cust-btn"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</section>
`;

exports[`Member List Item Component Rendering Tests renders to UI as expected with PROPS 1`] = `
<section
  className="main__content-user_lookup"
>
  <div
    className="cover"
  >
    <form
      className="customerLoginForm"
      onSubmit={[Function]}
    >
      <h2>
        View What Rewards You've Earned
      </h2>
      <div
        role="alert"
      />
      <div>
        <label
          htmlFor="phone"
        >
          Enter your phone number:
        </label>
        <input
          id="phone"
          name="phone_number"
          onChange={[Function]}
          pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
          placeholder="734-555-0505"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div>
        <button
          className="cust-btn"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</section>
`;
